namerd/storage/zk/src/main/scala/com/twitter/finagle/serverset2/client/buoyant/ZkSession.scala

```diff
@@ -123,71 +124,83 @@ class ZkSession(
       @volatile var closed = false
 
       def loop(): Future[Unit] = {
```

Comment:
This is so much more readable than what was here before!  :star:

Reply:
This method really is the heart of the zk watch code and is very semantically dense.  Perhaps adding some comments will help future spelunkers understand it more easily.

Reply:
Comments have been added to each stage of the chain of operations!